theano/scalar/basic.py

```diff
@@ -2399,9 +2399,19 @@ def c_code(self, node, name, inputs, outputs, sub):
 
 
 class Neg(UnaryScalarOp):
+    # We can use numpy.negative here, because even if it gives unexpected
+    # results on Boolean arrays, it will be passed other dtypes as Theano
+    # does not have a Boolean type for tensors.
     nfunc_spec = ('negative', 1, 1)
 
     def impl(self, x):
+        # We have to make sure x is not a numpy.bool_, because
+        # `-numpy.bool_(True)` is `False` (we want 0), and
+        # `-numpy.bool_(False)` is `True` (we want 1).
+        # This happens for Composite, as the intermediate results are not
+        # casted in the dtype of the intermediate variable in general.
+        if isinstance(x, numpy.bool_):
+            x = numpy.int8(x)
```

Comment:
It seem I missed that part. The other option would be to cast explicitly in each op that could output a bool. I don't like that better. We can merge when travis pass. A test would be great otherwise, but I don't think it is mandatory.

Which test was failing? I don't recall it. Was it only on one specific platform?